include/nuttx/wireless/ieee80211/ieee80211.h

```diff
@@ -42,849 +42,4070 @@
+{
+  uint8_t token;
+  uint8_t mode;
+  uint8_t type;
+  uint8_t request[];
+};
+
+/* struct ieee80211_channel_sw_ie
+ *
+ * This structure refers to "Channel Switch Announcement information element"
+ */
+
+struct ieee80211_channel_sw_ie
+{
+  uint8_t mode;
+  uint8_t new_ch_num;
+  uint8_t count;
+};
+
+/* struct ieee80211_ext_chansw_ie
+ *
+ * This structure represents the
+ * "Extended Channel Switch Announcement element"
+ */
+
+struct ieee80211_ext_chansw_ie
+{
+  uint8_t mode;
+  uint8_t new_operating_class;
+  uint8_t new_ch_num;
+  uint8_t count;
+};
+
+/* struct ieee80211_sec_chan_offs_ie - secondary channel offset IE
+ * @sec_chan_offs: secondary channel offset,
+ *                 uses IEEE80211_HT_PARAM_CHA_SEC_* values here
+ * This structure represents the "Secondary Channel Offset element"
+ */
+
+struct ieee80211_sec_chan_offs_ie
+{
+  uint8_t sec_chan_offs;
+};
+
+/* struct ieee80211_mesh_chansw_params_ie - mesh channel switch parameters IE
+ *
+ * This structure represents the "Mesh Channel Switch Paramters element"
+ */
+
+struct ieee80211_mesh_chansw_params_ie
+{
+  uint8_t mesh_ttl;
+  uint8_t mesh_flags;
+  uint16_t mesh_reason;
+  uint16_t mesh_pre_value;
+};
+
+/* struct ieee80211_wide_bw_chansw_ie - wide bandwidth channel switch IE
+ */
+
+struct ieee80211_wide_bw_chansw_ie
+{
+  uint8_t new_channel_width;
+  uint8_t new_center_freq_seg0;
+  uint8_t new_center_freq_seg1;
+};
+
+/* struct ieee80211_tim
+ *
+ * This structure refers to "Traffic Indication Map information element"
+ */
+
+struct ieee80211_tim_ie
+{
+  uint8_t dtim_count;
+  uint8_t dtim_period;
+  uint8_t bitmap_ctrl;
+
+  /* variable size: 1 - 251 bytes */
+
+  uint8_t virtual_map[1];
+};
+
+/* struct ieee80211_meshconf_ie
+ *
+ * This structure refers to "Mesh Configuration information element"
+ */
+
+struct ieee80211_meshconf_ie
+{
+  uint8_t meshconf_psel;
+  uint8_t meshconf_pmetric;
+  uint8_t meshconf_congest;
+  uint8_t meshconf_synch;
+  uint8_t meshconf_auth;
+  uint8_t meshconf_form;
+  uint8_t meshconf_cap;
+};
+
+/* enum mesh_config_capab_flags
+ * Mesh Configuration IE capability field flags
+ *
+ * @IEEE80211_MESHCONF_CAPAB_ACCEPT_PLINKS: STA is willing to establish
+ * additional mesh peerings with other mesh STAs
+ * @IEEE80211_MESHCONF_CAPAB_FORWARDING: the STA forwards MSDUs
+ * @IEEE80211_MESHCONF_CAPAB_TBTT_ADJUSTING: TBTT adjustment procedure
+ * is ongoing
+ * @IEEE80211_MESHCONF_CAPAB_POWER_SAVE_LEVEL:
+ *                        STA is in deep sleep mode or has
+ * neighbors in deep sleep mode
+ */
+
+enum mesh_config_capab_flags
+{
+  IEEE80211_MESHCONF_CAPAB_ACCEPT_PLINKS    = 0x01,
+  IEEE80211_MESHCONF_CAPAB_FORWARDING       = 0x08,
+  IEEE80211_MESHCONF_CAPAB_TBTT_ADJUSTING   = 0x20,
+  IEEE80211_MESHCONF_CAPAB_POWER_SAVE_LEVEL = 0x40,
+};
+
+#define IEEE80211_MESHCONF_FORM_CONNECTED_TO_GATE 0x1
+
+/* mesh channel switch parameters element's flag indicator */
+
+#define WLAN_EID_CHAN_SWITCH_PARAM_TX_RESTRICT  BIT(0)
```

Comment:
mix of `(1<<0)` and `BIT(0)` in this file

Reply:
Done